### PR TITLE
ensure other 2FA pages also handle session id

### PR DESCRIPTION
**Pull request by leohemsted**
specifically, the 2FA page when you first create an account is different to the login 2FA page
and also the 2FA page when you change your phone number is different as well

### CODE REVIEW

File: app/main/views/verify.py [hunk 1]
```diff
@@ -36,6 +36,9 @@ def _check_code(code):
     if form.validate_on_submit():
         try:
             user = user_api_client.get_user(user_id)
+            print(user.current_session_id)
```
Comment (quis) [inline app/main/views/verify.py]: 🖨

Comment (leohemsted) [inline app/main/views/verify.py]: :cold_sweat:

File: tests/app/main/views/test_user_profile.py [hunk 2]
```diff
@@ -204,29 +174,39 @@ def test_should_show_confirm_after_mobile_number_change(
...
     mock_check_verify_code,
+    fake_uuid
 ):
+    user_before = create_user(fake_uuid)
+    user_after = create_user(fake_uuid)
+    user_before.current_session_id = str(uuid.UUID(int=1))
+    user_after.current_session_id = str(uuid.UUID(int=2))
+
+    # first time (login decorator) return normally, second time (after 2FA return with new session id)
+    mocker.patch('app.user_api_client.get_user', side_effect=[user_before, user_after])
```
Comment (quis) [inline tests/app/main/views/test_user_profile.py]: 😎

File: tests/app/main/views/test_verify.py [hunk 3]
```diff
@@ -24,17 +26,28 @@ def test_should_return_verify_template(
...
         session['user_details'] = {'email_address': api_user_active.email_address, 'id': api_user_active.id}
+        # user's only just created their account so no session in the cookie
+        session.pop('current_session_id', None)
+
     response = client.post(url_for('main.verify'),
                            data={'sms_code': '12345'})
     assert response.status_code == 302
     assert response.location == url_for('main.add_service', first='first', _external=True)
 
+    # # make sure the current_session_id has changed to what the API returned
```
Comment (quis) [inline tests/app/main/views/test_verify.py]: Two `#`

`</pedantry>`

**Pull request merged**